jib-plugins-common/src/main/java/com/google/cloud/tools/jib/plugins/common/SkaffoldFilesOutput.java

```diff
@@ -0,0 +1,115 @@
+/*
+ * Copyright 2019 Google LLC.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package com.google.cloud.tools.jib.plugins.common;
+
+import com.fasterxml.jackson.annotation.JsonAutoDetect;
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.OutputStream;
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Builds a JSON string containing files and directories that <a
+ * href="https://github.com/GoogleContainerTools/skaffold">Skaffold</a> should watch for changes
+ * (and consequently trigger rebuilds).
+ *
+ * <p>{@code buildFiles} consist of build definitions. Changes in these files indicate that the
+ * project structure may have changed, so Skaffold will refresh the file watch list when this
+ * happens.
+ *
+ * <p>{@code inputs} consist of source/resource directories. Skaffold will trigger a rebuild when
+ * changes are detected in these files.
+ *
+ * <p>{@code ignore} consists of files that the Skaffold file watcher should not watch.
+ *
+ * <p>Example:
+ *
+ * <pre>{@code
+ * {
+ *   "buildFiles": [
+ *     "buildFile1",
+ *     "buildFile2"
+ *   ],
+ *   "inputs": [
+ *     "src/main/java/",
+ *     "src/main/resources/"
+ *   ],
+ *   "ignore": [
+ *     "pathToIgnore"
+ *   ]
+ * }
+ * }</pre>
+ */
+public class SkaffoldFilesOutput {
+
+  @JsonAutoDetect(fieldVisibility = JsonAutoDetect.Visibility.ANY)
+  @JsonIgnoreProperties(ignoreUnknown = true)
+  private static class SkaffoldFilesTemplate {
+
+    private final List<String> buildFiles = new ArrayList<>();
+
+    private final List<String> inputs = new ArrayList<>();
+
+    private final List<String> ignore = new ArrayList<>();
+  }
+
+  private final SkaffoldFilesTemplate skaffoldFilesTemplate = new SkaffoldFilesTemplate();
+
+  /**
+   * Adds a build file/directory.
```

Comment:
Is this actually build file/**directory**? If so, should we then rename `addBuildFile` -> `addBuild` ? Or are we mandating that this is all only files (and not directories?)

Reply:
@briandealwis left a comment on the design doc suggesting that we may want to include directories here as well as files:

>Do we need to be specific as to files-only, directories-only? Maven supports a .mvn/ directory where extensions can be put — these would be buildFiles. I could see -ignore- wildcards being useful for both input and ignore.